File: 75-sort-colors/sort-colors.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var sortColors = function(nums) {
    // merge
    function merge(left, right) {
        let [i, j] = [0, 0]
        const res = []
        while (i < left.length && j < right.length) {
            if (left[i] <= right[j]) {
                res.push(left[i])
                i += 1
            } else {
                res.push(right[j])
                j += 1
            }
        }
        while (i < left.length) {            
            res.push(left[i])
            i += 1
        }
        while (j < right.length) {
            res.push(right[j])
            j += 1
        }
        return res
    }
    // split
    function split(nums) {
        if (nums.length <= 1) return nums
        const m = Math.floor((nums.length - 1) / 2)
        const left = split(nums.slice(0, m + 1))
        const right = split(nums.slice(m + 1))
        return merge(left, right)
    }
    const res = split(nums)
    for (let i = 0; i < nums.length; i++) {
        nums[i] = res[i]
    }
};

File: 75-sort-colors/sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        def split(nums):
            if len(nums) <= 1: return nums
            mid = (0 + len(nums) - 1) // 2
            A = split(nums[:mid + 1])
            B = split(nums[mid + 1:])
            return merge(A, B)
        def merge(A, B):
            i, j = 0, 0
            res = []
            while i < len(A) and j < len(B):
                if A[i] <= B[j]:
                    res.append(A[i])
                    i += 1
                else:
                    res.append(B[j])
                    j += 1
            res += A[i:] + B[j:]
            return res
        nums[:] = split(nums)